app.rb

```diff
@@ -1,17 +1,21 @@
 require 'dotenv'
 require 'erb'
+require 'hashie/mash'
 require 'net/http'
+require 'pp'
 require 'sinatra/base'
 require 'onelogin/ruby-saml'
 require 'yaml'
 
 Dotenv.load
 
 class RelyingParty < Sinatra::Base
-  enable :sessions
+  use Rack::Session::Cookie, key: 'rack.session', path: '/', secret: SecureRandom.uuid
```

Comment:
I noticed this too and fixed it as part of #24.

I just used 'use Rack::Session::Cookie, :key => 'sinatra_sp' which seemed to be enough.  I wasn't sure exactly why this fix was needed but I thought it was perhaps that the 'rack.session' cookie was being clobbered by the several apps that were all running on 'localhost'.

Can you explain exactly why you think this fix is needed?